### PR TITLE
Implementation/stream service

### LINKED ISSUE #11: [FEATURE] Implement StreamService
**Feature description**
Need to implement all methods for IsStreamService interface. Path: `team.asd.service.IsStreamService`. Implemented service should be in package `team.asd.service`.

**Steps**
1. Checkout to provided branch
2. Implement provided interface according to javadoc description
3. Format your implemented code
4. Commit your changes with rich description
5. Check JUnit tests for your implementation
6. Push your commits to your repository
7. Create a Pull Request from working branch to `master` branch of your repository

**Expected result**
Fully implemented IsStreamService interface with successfully passed JUnit tests.

**Branch to work with**
Working branch: `implementation/StreamService`

### PR DESCRIPTION
**Pull request by stishchenko**

### CODE REVIEW

File: src/main/java/team/asd/service/StreamApiServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		return listOfDates.stream()
+				.filter(Objects::nonNull)
+				.sorted()
+				.filter(current -> current.isAfter(date) || current.isEqual(date))
+				.filter(curr -> dateFilter(curr, date, daysToSkip));
+	}
+
+	private boolean dateFilter(LocalDate current, LocalDate date, Integer skipper) {
+		if (current != null && date != null && skipper != null) {
+			if (current.isAfter(date) && skipper > 0) {
```
Comment (NckGrs) [inline src/main/java/team/asd/service/StreamApiServiceImpl.java]: combine conditions if possible to make fewer nested levels, also you can use utilities to simplify the code `ObjectUtils.allNotNull(current, date, skipper)`

File: src/main/java/team/asd/service/StreamApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+	@Override
+	@NonNull
+	public List<Integer> convertStringListToIntegerList(List<String> stringList) {
+		if (CollectionUtils.isEmpty(stringList)) {
+			return Collections.emptyList();
+		}
+		return stringList.stream()
+				.filter(Objects::nonNull)
+				.filter(str -> str.matches("-?[0-9]+"))
```
Comment (rdovgan) [inline src/main/java/team/asd/service/StreamApiServiceImpl.java]: @stishchenko you can remove this regex if you use `Integer.valueOf` method below

File: src/main/java/team/asd/service/StreamApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	@NonNull
+	public Stream<LocalDate> skipDaysFromSpecifiedDate(List<LocalDate> listOfDates, LocalDate date, Integer daysToSkip) {
+		if (CollectionUtils.isEmpty(listOfDates) || date == null || daysToSkip == null || daysToSkip < 0) {
+			return Stream.empty();
+		}
+		return listOfDates.stream()
+				.filter(Objects::nonNull)
+				.sorted()
+				.filter(current -> current.isAfter(date) || current.isEqual(date))
+				.filter(curr -> dateFilter(curr, date, daysToSkip));
```
Comment (rdovgan) [inline src/main/java/team/asd/service/StreamApiServiceImpl.java]: @stishchenko to skip some values you can use `Stream.skip` method

Comment (rdovgan) [reply src/main/java/team/asd/service/StreamApiServiceImpl.java:115]: 🤥 `limit`

**Review by rdovgan — APPROVED** [review]

**Pull request merged**